services/retentionmgr/retention.go

```diff
@@ -342,10 +435,9 @@ func (t *RetentionManager) runRetention(jobsC chan<- *retentionJob) bool {
 			totalJobs++
 		}
 
-		if allExtentsDeleted && dest.status == shared.DestinationStatus_DELETING {
+		if dest.status == shared.DestinationStatus_DELETING && numCGs == 0 && numExtents == 0 {
```

Comment:
I don't think you can rely on 'numCGs == 0' since the error from the below call was swallowed.
cgs := t.metadata.GetConsumerGroups(dest.id)

Reply:
I still don't see this get fixed.